log-group-events-lambda/handler/sf_ops.go

```diff
@@ -0,0 +1,169 @@
+package handler
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs/cloudwatchlogsiface"
+	"github.com/hashicorp/go-multierror"
+	"github.com/logzio/firehose-logs/common"
+	"os"
+	"sync"
+	"time"
+)
+
+type CloudWatchLogsClient struct {
+	Client cloudwatchlogsiface.CloudWatchLogsAPI
+}
+
+func getCloudWatchLogsClient() (*CloudWatchLogsClient, error) {
+	sess, err := common.GetSession()
+	if err != nil {
+		return nil, err
+	}
+	return &CloudWatchLogsClient{Client: cloudwatchlogs.New(sess)}, nil
+}
+
+func (cwLogsClient *CloudWatchLogsClient) addSubscriptionFilter(logGroups []string) ([]string, error) {
+	if cwLogsClient == nil {
+		return nil, fmt.Errorf("CloudWatch Logs client is nil")
+	}
+
+	destinationArn := os.Getenv(envFirehoseArn)
```

Comment:
Consider getting values from env during function initialisation

Reply:
Added envConfig which is initialized at the beginning of the function run and used instead of the `os.Getenv(xx)`